Monika-After-Story/MonikaModDev
# Deprecate Ren'Py logging and move to Python logging

`mas_log` and `MASMacLog` has been deprecated in favour of python2.7's `logging` library (which I have added to `python-packages` to port it. Ren'Py still uses an outdated version from python2.0)

# Major changes:
- Logging by default now doesn't require a `\n` to terminate lines
- Logging by default has severities which are denoted by methods on the `Logger` object, `debug`, `info`, `warning`, `error`, and `exception` are all supported and will be written to logs
- Logging by default will be ROTATING. As such, unless `append` is set to `False`, all logs will be appended and rotating to provide adequate debugging information over sessions (each log will have a max size of 5mb)
- Most config vars have been moved to `0config.rpy` from `options.rpy` to fix the issue of not-defined variables being used in logs
- `00utils.rpy` has been renamed to `0utils.rpy` as to avoid sharing a reserved name with internal renpy rpy's.

# Additions:
- 1 major function: `mas_logging.init_log`, which initializes a `logging.Logger` and --by default-- a `RotatingFileHandler`

# Testing:
- Verify all logs are working
- Verify new logs list severities appropriately
- Verify `aff_log.txt` appends as usual
- Verify `mas_log.txt` now appends always, regardless of unstable mode
- Verify `mfread.txt` and `mfgen.txt` logs work as expected and do NOT append
- Verify `spj.txt` does NOT append and lists info appropriately + has appropriate line-breaks
- Verify `submod_log.txt` does NOT append and lists all installed submods with their version numbers
- Verify `bg_flt.txt` does NOT append and lists info properly
